Send to server

ababiarz opened the pull request:
client, model: Chat-related models + sending/receiving messages from the global chat

ababiarz committed:
client, model: added sending messages to global chat mechanism

ababiarz committed:
model: chat functionality

ababiarz committed:
client: global messages sending in backend.service, auth.service and global-chat.component

ababiarz committed:
client: auth.service cleaned

jerry-sky committed:
model: cleaned up style of the code (@ababiarz)

jerry-sky commented on client/src/app/global-chat/global-chat.component.ts:
```diff
@@ -1,21 +1,32 @@
 })
 export class GlobalChatComponent implements OnInit {
-  messages: string[] = ['message1', 'message2', 'message3'];
+  username = 'benek';
```
Why is there a random hard-coded username?

jerry-sky commented on client/src/app/services/auth.service.ts:
```diff
@@ -87,4 +89,25 @@ export class AuthService {
+  }
 }
+
+  
```
Please fix this and other formatting issues (missing newlines at EOF, trailing whitespace and inappropriate line breaking).

jerry-sky commented on client/src/app/services/backend.service.ts:
```diff
@@ -16,6 +18,7 @@ export class BackendService {
   private usersUrl = environment.API + '/user';
   private mapUrl = environment.API + '/map';
   private cityUrl = environment.API + '/city';
+  private globalChatUrl = environment.API + '/globalChat';
```
For API routes and file names we use `caterpillar-case`, not `CamelCase`.

jerry-sky reviewed (changes requested):
Please fix these minor issues I have outlined below.

We will merge this branch back to `master` once the backend part of this functionality is ready (@jerry-sky, @Deatomizeer).

ababiarz committed:
client, model: retrieving chats from backend into messagebox; global-chat hard coded username fixed

ababiarz committed:
client, model: mostly added code clean-up

ababiarz committed:
client: removed weird things from chat.ts, sorry

ababiarz committed:
client, model: done code cleanup, at least i think so this time

jerry-sky committed:
client: moved all chat methods to a separate service (`ChatService`)

jerry-sky committed:
synced `send-to-server` with `master`

KamilMatejuk committed:
client: typo fix

ababiarz committed:
model, client: added private messages sending and retrieving; debugged user data retrieving

ababiarz commented:
Minor issues related to file formatting and file naming standard fixed.
Added retrieving username from backend.
Added private messages sending.

jerry-sky commented:
Roger that — I’ll try to implement this in the backend.

@Deatomizeer — we need to implement `model` classes related to chat into SQL tables.

jerry-sky commented:
Let’s assume that the global chat is chat of `id=1`.

jerry-sky committed:
model: added the global chat id

jerry-sky committed:
server: implemented global chat methods

Deatomizeer committed:
database: minor bug fixes in map generation

The procedure now generates the correct amount of cells, and their terrain type is always within the defined boundaries.

Deatomizeer committed:
database: added chat-related tables and a trigger

Note that the global chat has an ID of 1 instead of 0 for the time being.

Deatomizeer committed:
backend: added enum declarations for tables

Deatomizeer committed:
database: minor structure fix

Replaced the `global` flag in `channels` table to `private.

jerry-sky committed:
synced `send-to-server` with `backend-chat` (and thus with `master`)

Deatomizeer committed:
merged `backend-chat` into `send-to-server`

Deatomizeer committed:
backend: minor bug and style fixes in chat-related methods

Changed the Message model to reflect its database format and fixed the method for fetching the messages from the database.

ababiarz committed:
client added login/registration dialogs: error- and success-dialog

Deatomizeer committed:
database: expanded on the chat functionalities

Added a procedure responsible for creating private chatrooms between users, and allowed it to be called during a session.

Deatomizeer committed:
backend: added another error message

The error is thrown when trying to get messages from a nonexistent channel. Also replaced backticks with apostrophes in the `database-tables` enum

Deatomizeer committed:
backend: added methods for handling private conversations

Like with the global chat, a user can now send messages to private channels they\'re in and get the messages from the specified chatroom.

Deatomizeer committed:
Uh oh, we really do be merging right now.
Merging back-end chat implementation with some front-end changes.

jerry-sky commented on server/src/services/chat.service.ts:
```diff
@@ -35,4 +42,91 @@ export class ChatService {
+        [user1.id, user2.id, GLOBAL_CHAT_ID]
+      );
+    });
+    result = JSON.parse(JSON.stringify(result));
```
Wow wow wow, back up here a second — if you want to copy an object use `const copy = { ...original }`.

The second thing is that I don’t really understand why would you want to copy and assign the same object… to itself? Am I missing something here? I presume it’s a result of some debugging process :P.

Deatomizeer replied:
I don't know, it wouldn't work without that 70th line :(

jerry-sky replied:
Pushed a commit that resolves this issue.

jerry-sky committed:
server: fixed some minor variable declaration issue in method `ChatService.GetChannelId`

jerry-sky committed:
synced `send-to-server` with `master`

jerry-sky committed:
model: added a “user doesn’t exist” error

jerry-sky committed:
server: fixed the `/user/resources` route; selected user may not always exist

ababiarz committed:
client: got rid of unnecessary passing current user's username to server

ababiarz committed:
client: debugged sending messages to global/private chat

jerry-sky committed:
server: added `/user/usernames` route that returns a dictionary of all users with their usernames

ababiarz committed:
Merge branch 'send-to-server' of https://github.com/jerry-sky/cyber-city-sim into send-to-server

ababiarz committed:
client: added usernames and userids dictionary to show who wrote message in the global chat

ababiarz committed:
client: added refreshing global chat messages

ababiarz committed:
client: cleaned up code and added refreshing private chat messages

jerry-sky committed:
client: fixed client-side global chat refreshing

jerry-sky committed:
synced `send-to-server` with `master`

jerry-sky reviewed (approved):
It should be fine!
